src/Impersonator.php

```diff
@@ -158,13 +161,27 @@ protected function performStop(): void
         $guardName = $session->getGuardName();
 
         $guard = $this->auth->guard($guardName);
+
+        $recaller = $this->recaller($guard);
+
         /** @var Authenticatable $impersonatedUser */
         $impersonatedUser = $guard->user();
 
         $session->clear();
 
         $guard->logout();
-        $guard->loginUsingId($impersonatorId);
+
+        $remember = false;
+        if ($recaller instanceof Recaller && $recaller->valid()) {
+            $recallerId = $recaller->id();
+            if (is_int($impersonatorId)) {
+                $recallerId = intval($recallerId);
+            }
+
+            $remember = $impersonatorId === $recallerId;
```

Comment:
I think we could simplify this by using `$impersonatorId == $recaller->id()`, and we’d also get UUID/ULID support

```suggestion
            $remember = $impersonatorId == $recaller->id();
```